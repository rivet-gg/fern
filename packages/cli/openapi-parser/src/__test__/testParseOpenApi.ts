import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
