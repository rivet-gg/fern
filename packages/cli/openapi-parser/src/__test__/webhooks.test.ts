import { testParseOpenAPI } from "./testParseOpenApi";

describe("open api parser", () => {
    testParseOpenAPI("webhooks", "openapi.yml");
});
