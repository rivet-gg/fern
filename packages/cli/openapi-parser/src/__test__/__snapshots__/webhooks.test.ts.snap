// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open api parser webhooks parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [],
  "schemas": {
    "Pet": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "Pet",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "petId",
          "key": "id",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "PetId",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": null,
              "type": "int64",
            },
            "type": "primitive",
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "petName",
          "key": "name",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "PetName",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": null,
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "petTag",
          "key": "tag",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "petTag",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "PetTag",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": null,
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
      ],
      "type": "object",
    },
  },
  "securitySchemes": {},
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Webhooks",
  "variables": {},
  "webhooks": [
    {
      "description": null,
      "examples": [
        {
          "description": null,
          "name": "Example 1",
          "payload": {
            "type": "map",
            "value": [
              {
                "key": {
                  "type": "string",
                  "value": "id",
                },
                "value": {
                  "type": "primitive",
                  "value": {
                    "type": "int",
                    "value": 123,
                  },
                },
              },
              {
                "key": {
                  "type": "string",
                  "value": "name",
                },
                "value": {
                  "type": "primitive",
                  "value": {
                    "type": "string",
                    "value": "Scratch",
                  },
                },
              },
              {
                "key": {
                  "type": "string",
                  "value": "tag",
                },
                "value": {
                  "type": "primitive",
                  "value": {
                    "type": "string",
                    "value": "Cat",
                  },
                },
              },
            ],
          },
        },
      ],
      "generatedPayloadName": "CreatePetPayload",
      "headers": [],
      "method": "POST",
      "operationId": "CreatePet",
      "payload": {
        "availability": null,
        "description": null,
        "generatedName": "CreatePetPayload",
        "groupName": null,
        "nameOverride": null,
        "schema": "Pet",
        "type": "reference",
      },
      "sdkName": null,
      "summary": null,
      "tags": [],
    },
  ],
}
`;
